### PR TITLE
Add test to verify issue doesn't repro

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Closes https://github.com/dotnet/roslyn/issues/48006

### CODE REVIEW

**Review by Evangelink — APPROVED** [review]

File: src/Analyzers/CSharp/Tests/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs [hunk 1]
```diff
@@ -1966,5 +1966,46 @@ public async Task TopLevelStatement_FollowedWithThrow()
...
+        public async Task TestOnMultiCaseSection_String_CSharp9()
+        {
+            var testCode = @"
+class Program
+{
+    bool M(string s)
+    {
+        [|switch|] (s)
+        {
+	        case ""Last"":
```
Comment (allisonchou) [inline src/Analyzers/CSharp/Tests/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs:1981]: The formatting here looks a little off, although it looks like it was that way in the original too, so not sure if it's a big deal?

**Review by allisonchou — APPROVED** [review]

File: src/Analyzers/CSharp/Tests/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs (hunk 1 above)
Comment (allisonchou) [reply src/Analyzers/CSharp/Tests/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs:1981]: Oops nevermind, it looks fine now for some reason, guess something might have just been off earlier on my end.

Comment (allisonchou) [reply src/Analyzers/CSharp/Tests/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs:1981]: Oops nevermind, it looks fine now for some reason, guess something might have just been off earlier on my end.

File: src/Analyzers/CSharp/Tests/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs [hunk 2]
```diff
@@ -1966,5 +1966,46 @@ public async Task TopLevelStatement_FollowedWithThrow()
...
+            _ => false,
+        };
+    }
+}";
+
+            await new VerifyCS.Test
+            {
+                TestCode = testCode,
+                FixedCode = fixedCode,
+                LanguageVersion = CSharp9,
```
Comment (Youssef1313) [inline src/Analyzers/CSharp/Tests/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs:2007]: Does this need to be tested on earlier version?

Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/Tests/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs:2007]: i'm not sure what you mean.  can you clarify?

Comment (Youssef1313) [reply src/Analyzers/CSharp/Tests/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs:2007]: I meant the exact same test for C# 8 or any version before C# 9.

Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/Tests/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs:2007]: no.  'or' patterns aren't available in C# 8.

Comment (Youssef1313) [reply src/Analyzers/CSharp/Tests/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs:2007]: @CyrusNajmabadi Yes I know. I meant to confirm that the scenario in #48006 doesn't happen with previous version. (i.e. the test should probably check that no code fix is offerred at all for C# 8)

Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/Tests/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs:2007]: I'm satisfied this is covered with existing tests (like TestMissingOnMultiCaseSection).

**Pull request merged**